extensions/ql-vscode/src/qlpack-generator.ts

```diff
@@ -0,0 +1,84 @@
+import { mkdir, writeFile } from "fs-extra";
+import { dump } from "js-yaml";
+import { join } from "path";
+import { CodeQLCliServer } from "./cli";
+
+export type QueryLanguage =
+  | "csharp"
+  | "cpp"
+  | "go"
+  | "java"
+  | "javascript"
+  | "python"
+  | "ruby"
+  | "swift";
+
+export class QlPackGenerator {
+  private readonly qlpackName: string;
+  private readonly qlpackVersion: string;
+  private readonly header: string;
+  private readonly qlpackFileName: string;
+
+  constructor(
+    private readonly folderName: string,
+    private readonly queryLanguage: QueryLanguage,
+    private readonly cliServer: CodeQLCliServer,
+  ) {
+    this.qlpackName = `getting-started/codeql-extra-queries-${this.queryLanguage}`;
+    this.qlpackVersion = "1.0.0";
+    this.header = "# This is an automatically generated file.\n\n";
+
+    this.qlpackFileName = "qlpack.yml";
+  }
+
+  public async generate() {
+    await mkdir(this.folderName);
```

Comment:
For quick queries and adding database source archives, we use `workspace.updateWorkspaceFolders`. (E.g. [here](https://github.com/github/vscode-codeql/blob/465da69ecf41b58ef2050e66a23374e932105b23/extensions/ql-vscode/src/quick-query.ts#L43)).

We probably want something similar in this case 📂

Reply:
Thanks for thoroughly testing this! I've pushed up a commit to save the folder in the workspace instead of VS Code local storage.